### ISSUE TITLE
improve OSX compability

### ISSUE THREAD

**Issue by ulph** [clangd/vscode-clangd#157]
On OSX, clangd invokation can be achieved for instance via

```
#!/usr/bin/env bash
xcrun -r clangd "$@"
```

which will then be able to resolve system headers as well.

My suggestion is to probe for this route, installing some helper wrapper much like this, before suggesting to install and use a seperately built clangd.

**Comment by njames93** [comment]
Should point out I'm not an apple user.
It may be nice to have this automatically, but until then it should just be a case of calling xcrun with the find option and the copying the path returned to `clangd.path` in your vscode settings file.

**Comment by kadircet** [comment]
but i suppose this will run clangd shipped with xcode, right? which might be a quite older version, i am not sure if that would be beneficial for majority of users, as we actually have some logic to query xcrun and invoke clangd through right path to ensure it can find stdlib and such. (see https://github.com/llvm/llvm-project/blob/main/clang-tools-extra/clangd/CompileCommands.cpp#L31)

but it doesn't work in all cases(highly dependent on what you have in argv[0] of your compile flag), can you share clangd logs so that we can check why it is not working for you?

**Comment by sam-mccall** [comment]
Yeah, you're free to use Apple's version, but I wouldn't expect it to work particularly well, and we're completely unable to fix bugs in it on a useful timeline.

I *believe* the clangd in the latest xcode release (corresponding to swift 5.1) is built from approximately these sources, though the actual repo is not public:
https://github.com/apple/llvm-project/tree/swift/swift-5.1-branch/clang-tools-extra/clangd

Note that it's two years stale.

As Kadir points out, we do have logic to try to make this "just work" on mac, but not being mac people we're pretty reliant on detailed bug reports/suggestions for how to improve it/fix bugs.
